packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart

```diff
@@ -24,14 +23,19 @@ String? _getUrlScheme(String url) => Uri.tryParse(url)?.scheme;
 bool _isSafariTargetTopScheme(String url) =>
     _safariTargetTopSchemes.contains(_getUrlScheme(url));
 
+bool _navigatorIsSafari(html.Navigator navigator) =>
+    navigator.userAgent.contains('Safari') &&
+    navigator.vendor.contains('Apple Computer, Inc.') &&
```

Comment:
Out of curiosity, what case does this handle that the others don't?

Reply:
[MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent) explains how to detect browsers by sniffing the user agent string. For Safari, one has to check that the user agent string contains the word 'Safari', does not contain 'Chrome', and has  'Apple Computer, Inc.' as its vendor. This is to avoid false positives as Chrome sometimes reports as both Chrome and Safari.

Reply:
I understand why the !Chrome check is there, I was wondering about the vendor. Especially given that MDN docs [explicitly say not to use it](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/vendor).

Reply:
That's true. I used it because it was used in an [example implementation](https://github.com/flutter/engine/blob/0da06de991a90c169329d0d4788dcbcb16ebca0d/lib/web_ui/lib/src/engine/browser_detection.dart#L71-L72) I was looking at, and I missed its redundancy while testing. Thanks for the heads up.

Reply:
(This check is now based off of the user agent exclusively, resolving!)